base/applications/osk/main.c

```diff
@@ -785,6 +797,82 @@ LRESULT OSK_Paint(HWND hwnd)
 
     return 0;
 }
+
+/***********************************************************************
+ *
+ *           OSK_ThemeHandler
+ *
+ *  Function helper which handles theme drawing of buttons with icons.
+ */
+LRESULT APIENTRY OSK_ThemeHandler(LPNMCUSTOMDRAW pNmDraw)
+{
+    HTHEME hTheme;
+    HWND hButton;
+    INT iState = PBS_NORMAL;
+    HICON hIcon;
+    int x, y;
+
+    hButton = pNmDraw->hdr.hwndFrom;
+
+    /* If the button doesn't have an icon there is no need for custom handling */
+    if (!(GetWindowLongW(hButton, GWL_STYLE) & BS_ICON))
+        return CDRF_DODEFAULT;
+
+    /* Retrieve the theme handle for the button controls */
+    hTheme = GetWindowTheme(hButton);
+
+    if (!hTheme)
+        return CDRF_DODEFAULT;
+
+    /* Obtain CDDS drawing stages */
+    switch (pNmDraw->dwDrawStage)
+    {
+        case CDDS_PREPAINT:
+        {
+            /*
+                The button could be either in normal state or pushed.
+                Retrieve its state and save to a variable.
+            */
+            if (pNmDraw->uItemState & CDIS_DEFAULT)
+            {
+                iState = PBS_DEFAULTED;
+            }
+            else if (pNmDraw->uItemState & CDIS_SELECTED)
+            {
+                iState = PBS_PRESSED;
+            }
+            else if (pNmDraw->uItemState & CDIS_HOT)
```

Comment:
Just to be sure, it is impossible for the item to be _at the same time_ in `CDIS_SELECTED` _and_ `CDIS_HOT` states?

Reply:
The entire function was returned as a fix for button theming buttons.
I don't know if they can be in the same state.

Reply:
From my investigation it does not appear to be possible for an item to be `CDIS_SELECTED` and `CDIS_HOT`, at least in 2k3, 7, and 11.